api/v1_validators.inc

```diff
@@ -35,10 +35,6 @@ function validate_wordlist($wordlist, $data)
 
 function validate_page_name($pagename, $data)
 {
-    $pages = $data[":projectid"]->get_page_names_from_db();
```

Comment:
We can't neuter this validator -- this needs to still validate page names.

Reply:
Would it be better to validate it by querying the database directly rather than getting all the page names and checking if it is among them? I guess it woulfd be more efficient but we already have the function to get all the page  names. (Remebering what computers were like 50 years ago I tend to be unduly concerned with doing things most efficiently).

Reply:
In this case they're probably about the same cost. The image is a unique key in project table so it'll be an index-only read and that index is probably in memory already. It would be more efficient to query to see if a specific page exists if you wanted to add such a function:
```
mysql> explain select image from projectID5e26579e08db8;
+----+-------------+------------------------+------------+-------+---------------+-------+---------+------+------+----------+-------------+
| id | select_type | table                  | partitions | type  | possible_keys | key   | key_len | ref  | rows | filtered | Extra       |
+----+-------------+------------------------+------------+-------+---------------+-------+---------+------+------+----------+-------------+
|  1 | SIMPLE      | projectID5e26579e08db8 | NULL       | index | NULL          | image | 50      | NULL |  420 |   100.00 | Using index |
+----+-------------+------------------------+------------+-------+---------------+-------+---------+------+------+----------+-------------+
1 row in set, 1 warning (0.00 sec)

mysql> explain select image from projectID5e26579e08db8 where image = '420.png';
+----+-------------+------------------------+------------+-------+---------------+-------+---------+-------+------+----------+-------------+
| id | select_type | table                  | partitions | type  | possible_keys | key   | key_len | ref   | rows | filtered | Extra       |
+----+-------------+------------------------+------------+-------+---------------+-------+---------+-------+------+----------+-------------+
|  1 | SIMPLE      | projectID5e26579e08db8 | NULL       | const | image         | image | 50      | const |    1 |   100.00 | Using index |
+----+-------------+------------------------+------------+-------+---------------+-------+---------+-------+------+----------+-------------+
1 row in set, 1 warning (0.00 sec)

```